cli/awsbatch/awsbsub.py

```diff
@@ -465,6 +473,9 @@ def run(  # noqa: C901 FIXME
                 container_overrides.update(vcpus=vcpus)
             if memory:
                 container_overrides.update(memory=memory)
+            if gpus:
+                self.__validate_instance_type()
```

Comment:
Can't we just let the boto3 batch client perform the validations and fail on the API call instead of having them explicitly performed in our code? Otherwise we cannot guarantee to be aligned with batch requirements and will be hard to update our validations whenever batch does that